pkg/generate/app/sourcelookup.go

```diff
@@ -15,6 +16,11 @@ import (
 	s2igit "github.com/openshift/source-to-image/pkg/scm/git"
 )
 
+var (
+	argumentGit         = regexp.MustCompile("^(http://|https://|git@|git://).*(?:#([a-zA-Z0-9]*))?$")
+	argumentGitProtocol = regexp.MustCompile("^(git@|git://)")
```

Comment:
url.Parse *shouldn't* choke on it, but from my experience, it chokes on a few things it shouldn't choke on ;-)

i would still add the failing URL example to the git_test.go file I mentioned, and debug there to make sure the expected validation decision is occurring vs. adding the precursor checking in sourcelookup.go